test/Encoder/RailFenceCipher.js

```diff
@@ -0,0 +1,31 @@
+import { describe } from 'mocha'
+
+import EncoderTester from '../Helper/EncoderTester'
+import RailFenceCipherEncoder from '../../src/Encoder/RailFenceCipher'
+
+/** @test {RailFenceCipherEncoder} */
+describe('RailFenceCipherEncoder', () => EncoderTester.test(RailFenceCipherEncoder, [
+  {
+    // Wikipedia example
+    settings: { key: 3 },
+    content: 'WEAREDISCOVEREDFLEEATONCE',
+    expectedResult: 'WECRLTEERDSOEEFEAOCAIVDEN'
+  },
+  {
+    settings: { key: 25 },
+    content: 'WEAREDISCOVEREDFLEEATONCE',
+    expectedResult: 'WEAREDISCOVEREDFLEEATONCE'
+  },
+  {
+    settings: { key: 2 },
+    content: 'WEAREDISCOVEREDFLEEATONCE',
+    expectedResult: 'WAEICVRDLETNEERDSOEEFEAOC'
+  },
+  {
+    // if key is less than 2 then it will be set to default i.e. 2
+    // therefore even if key = 1, it's output is same as the case with key = 2
+    settings: { key: 1 },
+    content: 'WEAREDISCOVEREDFLEEATONCE',
+    expectedResult: 'WAEICVRDLETNEERDSOEEFEAOC'
+  }
```

Comment:
The result of an invalid input should be an error. When using the `min` constraint on the number setting as mentioned above it throws the error for you, no further tests needed here.